src/kontrol/prove.py

```diff
@@ -902,6 +924,60 @@ def _create_initial_account_list(
     return init_account_list
 
 
+def _create_cse_accounts(
+    foundry: Foundry, storage_fields: tuple[StorageField, ...], contract_name: str, contract_storage: KApply
+) -> tuple[list[KApply], list[KApply], KApply]:
+    # TODO: call this function recursively
+    new_accounts = []
+    new_account_constraints = []
+
+    for field in storage_fields:
+        if field.data_type.startswith('contract '):
+            contract_type = field.data_type.split(' ')[1]
+            for contract_name, contract_obj in foundry.contracts.items():
+                if contract_name.split('%')[-1] == contract_type:
+                    contract_code = KEVM.bin_runtime(KApply(f'contract_{contract_obj.name_with_path}'))
+                    contract_account_name = 'CONTRACT-' + field.label.upper()
+                    new_account = Foundry.symbolic_account(contract_account_name, contract_code)
+                    new_accounts.append(new_account)
+
+                    # TODO: handle the case where the field has a non-zero offset
+                    contract_storage = KApply(
+                        'Map:update',
+                        contract_storage,
+                        intToken(field.slot),
+                        KVariable(contract_account_name + '_ID'),
+                    )
```

Comment:
It slipped my mind that these could also be offset arbitrarily within a slot. With that knowledge, I think that this should be `#lookup` constraint, just like @anvacaru has been creating in his related PRs.

Reply:
Created a separate issue for it to be addressed in a follow-up PR: https://github.com/runtimeverification/kontrol/issues/622.